### PR TITLE
Site preview images and some other metadata bits

### PR DESCRIPTION
**Pull request by omgmog**
This should give us working `/preview/XYZ.png` previews for use on Twitter and other social shares.

These are generated automatically 🎉 

home

code-of-conduct

This will make tweets like this: https://twitter.com/JSOxford/status/1359148138204966916 look a  lot COOLER 😎

### CODE REVIEW

**Comment by omgmog** [comment]
Screenshot 2021-02-15 at 17 53 45

:boom:

**Comment by omgmog** [comment]
Looks like the meta descriptions could do with an added `| safe` to handle entity encoding better 🤔 

Don't merge this until I've pushed a commit for that

File: src/_includes/layouts/base.njk [hunk 1]
```diff
@@ -1,42 +1,44 @@
...
+      name="twitter:title"
+      content="{{pageTitle}}"
     />
 
+    {%- set pageDescription %}{% if description %}{{ description }}{% else %}{{ meta.siteDescription }}{% endif %}{% endset -%}
     <meta
-      name="og:title"
-      content="{{pageTitle}}"
+      name="description"
+      content="{{pageDescription}}"
```
Comment (omgmog) [inline src/_includes/layouts/base.njk:28]: `| safe` needed here

File: src/_includes/layouts/base.njk [hunk 2]
```diff
@@ -1,42 +1,44 @@
...
     />
     <meta
       name="og:description"
       content="{{pageDescription}}"
     />
-    <meta property="og:image" content="{{pageSocialImg}}" />
-    <meta name="twitter:image" content="{{pageSocialImg}}" />
+    <meta
+      name="twitter:description"
+      content="{{pageDescription}}"
```
Comment (omgmog) [inline src/_includes/layouts/base.njk:36]: `| safe` needed here

File: src/_includes/layouts/base.njk [hunk 3]
```diff
@@ -1,42 +1,44 @@
...
     <link rel="icon" href="{{ '/favicon.png' | url }}" />
     <link href="{{ '/css/style.css' | url }}" rel="stylesheet" />
+
     <meta name="twitter:card" content="summary_large_image" />
     {% if meta.twitterUsername %}
     <meta name="twitter:creator" content="@{{ meta.twitterUsername }}" />
     {% endif %}
+    
+    {%- set pageTitle %}{% if title %}{{ title }} | {% else %}{% if description %}{{ description | safe}} | {% endif %}{% endif %}{{ meta.siteName }}{% endset -%}
+    <title>{{pageTitle}}</title>
```
Comment (omgmog) [inline src/_includes/layouts/base.njk:15]: `| safe` needed here

File: src/_includes/layouts/base.njk [hunk 4]
```diff
@@ -1,42 +1,44 @@
...
     <meta
-      name="twitter:title"
+      name="og:title"
       content="{{pageTitle}}"
     />
     <meta
-      name="twitter:description"
-      content="{{pageDescription}}"
+      name="twitter:title"
+      content="{{pageTitle}}"
```
Comment (omgmog) [inline src/_includes/layouts/base.njk:22]: `| safe` needed here

**Comment by omgmog** [comment]
Oh I did not mean to push that, bare with :100:

**Comment by omgmog** [comment]
I think this is ready for a review 👍

**Review by javorszky — APPROVED** [review]

**Pull request merged**